src/v/raft/recovery_memory_quota.cc

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "raft/recovery_memory_quota.h"
+
+#include "raft/logger.h"
+#include "vlog.h"
+
+#include <seastar/core/memory.hh>
+
+namespace raft {
+
+recovery_memory_quota::recovery_memory_quota(
+  recovery_memory_quota::config_provider_fn config_provider)
+  : _cfg(config_provider())
+  , _current_max_recovery_mem(
+      _cfg.max_recovery_memory().value_or(default_max_memory()))
+  , _memory(_current_max_recovery_mem) {
+    _cfg.max_recovery_memory.watch([this] { on_max_memory_changed(); });
+}
+
+ss::future<ss::semaphore_units<>> recovery_memory_quota::acquire_read_memory() {
+    return ss::get_units(
+      _memory,
+      std::min(_current_max_recovery_mem, _cfg.default_read_buffer_size()));
+}
+
+size_t recovery_memory_quota::default_max_memory() {
+    return 0.15 * ss::memory::stats().total_memory();
+}
```

Comment:
should this be in `resource_mgmt/memory_groups.h` ?

Reply:
good idea, i will move it there